website/docs/r/cdn_frontdoor_rule.html.markdown

```diff
@@ -0,0 +1,702 @@
+
+* `match_values` - (Required) One or more string values representing the value of the request hostname to match. If multiple values are specified, they're evaluated using `OR` logic.
+
+* `transform` - (Optional)	Any transform from the `Condition Transform List` as defined below. Possible values include `Lowercase` , `RemoveNulls`, `Trim`, `Uppercase`, `UrlDecode` or `UrlEncode`. Defaults to `Lowercase`.
+
+---
+
+A `server_port_condition` block supports the following:
+
+The `server_port_condition` identifies requests based on which port of the Frontdoor server accepted the request on.
+
+* `operator` - (Required)	Any operator from the `Condition Operator list` as defined below. Possible values include `Any`, `Equal`, `Contains`, `BeginsWith`, `EndsWith`, `LessThan`, `LessThanOrEqual`, `GreaterThan`, `GreaterThanOrEqual` or `RegEx`.
+
+* `negate_condition` - (Optional) If `true` operator becomes the opposite of its value, for more information see `Condition Operator List` as defined below. Possible values `true` or `false`. Defaults to `false`.
+
+* `match_values` - (Required) One or more integer values(e.g. "1") representing the value of the client port to match. Possible values include `80` or `443`. If multiple values are specified, they're evaluated using `OR` logic.
+
+---
+
+A `client_port_condition` block supports the following:
+
+The `client_port_condition` identifies requests based on the port of the client which made the request.
+
+* `operator` - (Required)	Any operator from the `Condition Operator list` as defined below. Possible values include `Any`, `Equal`, `Contains`, `BeginsWith`, `EndsWith`, `LessThan`, `LessThanOrEqual`, `GreaterThan`, `GreaterThanOrEqual` or `RegEx`.
+
+* `negate_condition` - (Optional) If `true` operator becomes the opposite of its value, for more information see `Condition Operator List` as defined below. Possible values `true` or `false`. Defaults to `false`.
+
+* `match_values` - (Optional) One or more integer values(e.g. "1") representing the value of the client port to match. If multiple values are specified, they're evaluated using `OR` logic.
+
+---
+
+A `socket_address_condition` block supports the following:
+
+The `socket_address_condition` identifies requests based on the IP address of the direct connection to the Frontdoors edge. If the client used an HTTP proxy or a load balancer to send the request, the value of Socket address is the IP address of the proxy or load balancer.
+
+Remote Address represents the original client IP that is either from the network connection or typically the `X-Forwarded-For` request header if the user is behind a proxy.
+
+* `operator` - (Optional) The type of match. The Possible value is `IpMatch`. Defaults to `IpMatch`.
```

Comment:
I know this is very confusing, but they are conditionally `Required` depending on what the `operator` is set as. If the `operator` is `Any`, the `match_values` are not needed since it will match all IP addresses, if the `operator` is set to `IPMatch` then the `match_values` are `Required` to have at least one value. It is enforced in the `ExpandCdnFrontDoorSocketAddressCondition` function.